include/flexflow/model.h

```diff
@@ -638,9 +638,37 @@ class FFModel {
     return this->new_node(op);
   }
 
+  template <typename T>
+  PCG::Node get_or_create_node(const std::vector<ParallelTensor>& inputs, typename T::Params const& params) {
+    using Params = typename T::Params;
+
+    ParallelTensorShapes input_shapes;
+    for (const auto& input : inputs) {
+      input_shapes.push_back(input->get_shape());
+    }
+
+    T* op = nullptr;
+
+    std::pair<ParallelTensorShapes, Params> key{input_shapes, params};
+    auto &cache = this->get_cache_multi_inputs<T>();
+    const auto &it = cache.find(key);
+    if (it != cache.end()) {
+      op = it->second;
+    } else {
+      op = new T(*this, params, inputs, NULL);
+      cache[key] = op;
+    }
+
+    assert(op->get_params() == params);
+    return this->new_node(op);
+  }
+
   template <typename T>
   std::unordered_map<std::pair<ParallelTensorShape, typename T::Params>, T*> &get_cache();
 
+  template <typename T>
+  std::unordered_map<std::pair<ParallelTensorShapes, typename T::Params>, T*> &get_cache_multi_inputs();
```

Comment:
Along with the proposed change to `get_or_create_node`, here convert to 
```suggestion
  std::unordered_map<std::pair<typename T::Inputs, typename T::Params>, T*> &get_cache_inputs();
```